### PR TITLE
ENG-12024 expression index behaves strangely when divided by 0.

### PR DESCRIPTION
**Pull request by yzhang1991**

### CODE REVIEW

File: src/frontend/org/voltdb/types/ExpressionType.java [hunk 1]
```diff
@@ -58,15 +58,15 @@
         // left / right (both must be number. implicitly casted)
     OPERATOR_CONCAT                (OperatorExpression.class,  5, "||"),
         // left || right (both must be char/varchar)
-    OPERATOR_MOD                   (OperatorExpression.class,  6, "%"),
+    OPERATOR_MOD                   (OperatorExpression.class,  6, "%", true),
```
Comment (BillWhite) [inline src/frontend/org/voltdb/types/ExpressionType.java]: What if the expression is mod(a, b) and b is zero?  Isn't this undefined?

File: src/frontend/org/voltdb/types/ExpressionType.java [hunk 2]
```diff
@@ -116,8 +116,8 @@
     AGGREGATE_SUM                 (AggregateExpression.class, 42, "SUM"),
     AGGREGATE_MIN                 (AggregateExpression.class, 43, "MIN", true),
     AGGREGATE_MAX                 (AggregateExpression.class, 44, "MAX", true),
-    AGGREGATE_AVG                 (AggregateExpression.class, 45, "AVG"),
-    AGGREGATE_APPROX_COUNT_DISTINCT(AggregateExpression.class, 46, "APPROX_COUNT_DISTINCT"),
+    AGGREGATE_AVG                 (AggregateExpression.class, 45, "AVG", true),
```
Comment (BillWhite) [inline src/frontend/org/voltdb/types/ExpressionType.java]: Isn't AVG = SUM()/COUNT()?  What if the SUM overflows while calculating the average?

File: tests/frontend/org/voltdb/TestAdhocCreateDropIndex.java [hunk 3]
```diff
@@ -308,6 +308,92 @@ public void testCreateDropIndexonView() throws Exception
...
+            ENG12024TestHelper(ddlTemplateForBooleanExpression, "a = b", true);
+            ENG12024TestHelper(ddlTemplateForBooleanExpression, "a <> b", true);
+            ENG12024TestHelper(ddlTemplateForBooleanExpression, "a < b", true);
+            ENG12024TestHelper(ddlTemplateForBooleanExpression, "a > b", true);
+            ENG12024TestHelper(ddlTemplateForBooleanExpression, "a <= b", true);
+            ENG12024TestHelper(ddlTemplateForBooleanExpression, "a >= b", true);
+            ENG12024TestHelper(ddlTemplateForBooleanExpression, "c LIKE 'abc%'", true);
+            ENG12024TestHelper(ddlTemplateForBooleanExpression, "a IS NOT DISTINCT FROM b", true);
+            ENG12024TestHelper(ddlTemplateForBooleanExpression, "a > b AND b > 0", true);
+            ENG12024TestHelper(ddlTemplateForBooleanExpression, "a > b OR b > 0", true);
```
Comment (wolffcm) [inline tests/frontend/org/voltdb/TestAdhocCreateDropIndex.java]: What about functions?  Some are safe some are not.  I forget what we do for views in this case.

Comment (yzhang1991) [inline tests/frontend/org/voltdb/TestAdhocCreateDropIndex.java]: Function expressions are considered unsafe. We coded this, but I will add a test case here.

**Pull request merged**